### PR TITLE
[JIT] Refactor SR invocation of fusion

### PR DESCRIPTION
**Pull request by eellison**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #70656
* #70464
* #70871
* __->__ #70508
* #70463

We can create the code object at compile time instead or runtime to speed it up. This also makes unnecessary the compilation cache. TODO: figure out if theres a way to cache InterpreterState object

Differential Revision: [D33458648](https://our.internmc.facebook.com/intern/diff/D33458648)

### CODE REVIEW

File: torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp [hunk 1]
```diff
@@ -388,21 +388,21 @@ RegisterOperators reg_guard({
...
-  Code code(graph, "");
+void runTensorExprDynamicGroup(const Code& code, Stack& stack) {
   InterpreterState interpreter{code};
   interpreter.run(stack);
 }
 
 Operation createTensorExprDynamicGroup(const Node* node) {
   auto graph = node->g(attr::Subgraph);
+  Code code(graph, "");
   // This implementation creates a Code object and InterpreterState on every
```
Comment (navahgar) [inline torch/csrc/jit/passes/symbolic_shape_runtime_fusion.cpp:399]: Update this comment to reflect this change.

**Review by navahgar — APPROVED** [review]
Thanks for these changes.

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458648).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458648).

**Comment by eellison** [comment]
@eellison has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D33458648).

**Pull request closed without merging**